src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,12 +1,174 @@
 {
-    "$defs": {
-        "dateTimePattern": {
-            "type": "string",
-            "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2})$"
+  "$defs": {
+    "metatask": {
+      "type": "object",
+      "properties": {
+        "var": {
+          "$ref": "#/$defs/var"
         }
+      },
+      "patternProperties": {
+        "^task(_[a-z0-9_]+)?$": {
+          "$ref": "#/$defs/task"
+        },
+        "^metatask(_[a-z0-9_]+)?$": {
+          "$ref": "#/$defs/metatask"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "var"
+      ],
+      "minProperties": 2,
+      "maxProperties": 3
+    },
+    "task": {
+      "type": "object",
+      "properties": {
+        "attrs": {
+            "maxtries": {
+                "type": "string",
+                "format": "number",
+                "minimum": 0
+            },
+            "throttle": {
+                "type": "integer",
+                "minimum": 0
+            },
+            "cycledefs": {
+                "type": "string"
+            },
+            "final": {
+                "type": "boolean"
+            }
+        },
+        "command": {
+          "type": "string"
+        },
+        "cores": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "nodes": {
+          "type": "string"
+        },
+        "native": {
+            "type": "string"
+        },
+        "walltime": {
+          "$ref": "#/$defs/timestr"
+        },
+        "account": {
+            "type": "string"
+        },
+        "jobname": {
+            "type": "string"
+        },
+        "queue": {
+            "type": "string"
+        },
+        "partition": {
+            "type": "string"
+        },
+        "memory": {
+            "type": "string"
+        },
+        "rewind": {
+            "type": "string"
+        },
+        "envar": {
+            "type": "object"
+        },
+        "shared": {
+            "type": "string"
+        },
+        "exclusive": {
+            "type": "string"
+        },
+        "stdout": {
+            "type": "string"
+        },
+        "stderr": {
+            "type": "string"
+        },
+        "join": {
+            "type": "string"
+        },
+        "deadline": {
+            "$ref": "#/$defs/datePattern"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "command",
+        "walltime"
+      ],
+      "oneOf": [
+        {
+            "required": ["cores"]
+        },
+        {
+            "required": ["nodes"]
+        },
+        {
+            "required": ["native"]
+        }
+      ],
+      "anyOf": [
+        {
+            "not": {
+                "required": ["shared"]
+            }
+        },
+        {
+            "not": {
+                "required": ["exclusive"]
+            }
+        }
+      ],
```

Comment:
Is this `anyOf` + `not` pattern how we specify mutually exclusive but also optional properties? As in, "You don't have to specify either of these, but if you specify one you cannot specify the other"?

Reply:
Yes it is! Exactly. We don't need either `shared` or `exclusive`, but if one is supplied the other cannot be. Take a look at `stderr`, `stdout` and `join`. These 3 properties are not rewuired either, but in this case `stderr` and `stdout` have to show up as a pair if one or the other is supplied, and cannot appear with `join`. So we have a `stderr` AND `stdout` OR `join` situation. I did this with an added `dependencies` on `stdout` and `stderr`.